hazelcast/src/test/java/com/hazelcast/client/ClientRegressionWithRealNetworkTest.java

```diff
@@ -59,6 +62,41 @@
 @Category(SlowTest.class)
 public class ClientRegressionWithRealNetworkTest extends ClientTestSupport {
 
+    private static final Random rnd = new Random();
+
+    private static int getRandomAvailablePort() {
+        int port = rnd.nextInt(65535) + 1;
+        if (TestUtil.isPortAvailable(port)) {
+            return port;
```

Comment:
This will most probably work fine most of the time, but it is theoretically possible that the port might be occupied the instant you try to return it.

May I suggest using random cluster names in the tests? I believe this way, the tests will be much more simple, and we don't have to deal with the ports. I believe these tests never failed to use the default ports, but failed to the unknown join requests, so we should be fine using the default ports.